module/VuFind/src/VuFind/ILS/Driver/Folio.php

```diff
@@ -849,6 +873,25 @@ public function getHolding($bibId, array $patron = null, array $options = [])
                 }
                 $nextBatch[] = $nextItem;
             }
+
+            // If there are no item records at this location, we're going to create a fake one,
+            // fill it with data from the FOLIO holdings record, and make it not appear in
+            // the full record display using a non-visible AvailabilityStatus.
+            if ($number == 0) {
+                $locAndHoldings = $this->getItemFieldsFromLocAndHolding($holding->effectiveLocationId, $holdingDetails);
+                $invisibleAvailabilityStatus = new AvailabilityStatus(
+                    true,
+                    'HoldingStatus::holding_no_items_availability_message'
+                );
+                $invisibleAvailabilityStatus->setVisibility(false);
+                $nextBatch[] = $locAndHoldings + [
```

Comment:
There an (intentional, I think) inconsistency here where normal items are sent through `$this->formatHoldingItem()` and that output added to `$nextBatch`, whereas for the generated/fake item there is no format step.  It makes sense to not 'format' something that will not be visually displayed.  But the inconsistency bugs me, maybe because there could be use cases (not covered here) for actually displaying these generated items -- I think Lehigh did that until recently in a similar scenario.  One option would be to just extract the entire contents of the `if ($number == 0)` block into a helper method, for that to be overridden as needed.  But of course that could be a later enhancement and isn't necessary at this point.

Reply:
I haven't been deep enough into VuFind to understand the inconsistency...which is to say, I'm not sure what kind of "format" action is taking place...it looks like it is filling the contents of VuFind's item array with the response from FOLIO's API, and this refactoring isn't substantially different.

I thought about pulling out the contents of `if ($number == 0)` into a separate method, but I couldn't think of any use cases where that would be useful. Happy to make the change if y'all thing it would be wise.

Reply:
I'd say, ignore this suggestion and continue as-is.  If I come up with a concrete use case later on, it's an easy enough refactoring to do then.